test/base/test_concurrency_py3k.py

```diff
@@ -101,3 +107,35 @@ def go():
             await greenlet_spawn(go)
 
         await to_await
+
+    @async_test
+    @testing.requires.python37
+    async def test_contextvars(self):
+        import asyncio
+        import contextvars
+
+        var = contextvars.ContextVar("var")
+        event = asyncio.Event()
+        counter = [0]
+        concurrency = 5
+
+        async def async_inner(val):
+            eq_(val, var.get())
+
```

Comment:
**mike bayer** (zzzeek) wrote:

I tried to make this test simpler and easier to understand.  if I run the latest git master of greenlet, the test passes unless I take out the new contextvar logic in which case it fails, which suggests that this test is accurately testing the behavior.  if you can confirm i am not missing anything that would be helpful.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2376

Reply:
**Fantix King** (fantix) wrote:

Yes excactly, that's been my way to verify that the test is working correctly, by commenting out the "self.gr_context = _copy_context()" in the greenlet subclass and watch the test fails.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2376